tests/ecmp/test_fgnhg.py

```diff
@@ -0,0 +1,333 @@
+            port_list.append(ptf_port_id)
+            eth_port_list.append(port)
+            index = index + 1
+
+    port_list.sort()
+    bank_0_port = port_list[:len(port_list)/2]
+    bank_1_port = port_list[len(port_list)/2:]
+
+    # Create vlan if
+    duthost.command('config interface ip add Vlan' + str(DEFAULT_VLAN_ID) + ' ' + str(vlan_ip))
+
+    for index, ip in enumerate(vlan_ip.hosts()):
+        if len(ip_to_port) == NUM_NHs:
+            break
+        ip_to_port[str(ip)] = port_list[index]
+
+    return port_list, ip_to_port, bank_0_port, bank_1_port
+
+
+def generate_fgnhg_config(duthost, ip_to_port, bank_0_port, bank_1_port, prefix):
+    if isinstance(ipaddress.ip_network(prefix), ipaddress.IPv4Network):
+        fgnhg_name = 'fgnhg_v4'
+    else:
+        fgnhg_name = 'fgnhg_v6'
+
+    fgnhg_data = {}
+
+    fgnhg_data['FG_NHG'] = {}
+    fgnhg_data['FG_NHG'][fgnhg_name] = {
+        "bucket_size": 125
+    }
+
+    fgnhg_data['FG_NHG_MEMBER'] = {}
+    for ip, port in ip_to_port.items():
+        bank = 0
+        if port in bank_1_port:
+            bank = 1
+        fgnhg_data['FG_NHG_MEMBER'][ip] = {
+            "bank": bank,
+            "FG_NHG": fgnhg_name
+        }
+
+    fgnhg_data['FG_NHG_PREFIX'] = {}
+    fgnhg_data['FG_NHG_PREFIX'][prefix] = {
+        "FG_NHG": fgnhg_name
+    }
+
+    duthost.copy(content=json.dumps(fgnhg_data, indent=2), dest="/tmp/fgnhg.json")
+    duthost.shell("sonic-cfggen -j /tmp/fgnhg.json --write-to-db")
+
+
+def setup_neighbors(duthost, ptfhost, ip_to_port):
+    vlan_name = "Vlan"+ str(DEFAULT_VLAN_ID)
+    neigh_entries = {}
+    neigh_entries['NEIGH'] = {}
+
+    for ip, port in ip_to_port.items():
+
+        if isinstance(ipaddress.ip_address(ip.decode('utf8')), ipaddress.IPv4Address):
+            neigh_entries['NEIGH'][vlan_name + "|" + ip] = {
+                "neigh": ptfhost.shell("cat /sys/class/net/eth" + str(port) + "/address")["stdout_lines"][0],
+                "family": "IPv4"
+            }
+        else:
+            neigh_entries['NEIGH'][vlan_name + "|" + ip] = {
+                "neigh": ptfhost.shell("cat /sys/class/net/eth" + str(port) + "/address")["stdout_lines"][0],
+                "family": "IPv6"
+            }
+
+    duthost.copy(content=json.dumps(neigh_entries, indent=2), dest="/tmp/neigh.json")
+    duthost.shell("sonic-cfggen -j /tmp/neigh.json --write-to-db")
+
+
+def create_fg_ptf_config(ptfhost, ip_to_port, port_list, bank_0_port, bank_1_port, router_mac, net_ports, prefix):
+    fg_ecmp = {
+            "ip_to_port": ip_to_port,
+            "port_list": port_list,
+            "bank_0_port": bank_0_port,
+            "bank_1_port": bank_1_port,
+            "dut_mac": router_mac,
+            "dst_ip": prefix.split('/')[0],
+            "net_ports": net_ports,
+            "inner_hashing": USE_INNER_HASHING,
+            "num_flows": NUM_FLOWS 
+    }
+    ptfhost.copy(content=json.dumps(fg_ecmp, indent=2), dest=FG_ECMP_CFG)
```

Comment:
@anish-n can we add some logging here (`json.dumps(fg_ecmp, indent=2)`) just to facilitate the debugging?